### ISSUE TITLE
YAML export does not export proper YAML or correctly spelled "starred" field

### ISSUE THREAD

**Issue by micahellison** [jrnl-org/jrnl#835]
## Bug report

  - Jrnl version: 2.2
  - How you installed Jrnl: pip install jrnl
  - Operating system [MacOS, Linux, Windows?]: Windows 10

### Steps to repro:
```
jrnl --export yaml -o .\jrnltest
```

### Expected results:
 * Export should be in YAML format
 * Fields should be spelled correctly

### Actual results:
 * The output has YAML in it, but the entry contents are not encapsulated in a YAML field; instead there's just a space before the entry.
 * The "starred" field is spelled "stared" 
```
title: Test
date: 2020-02-08 12:05
stared: False
tags:
 So, I was testing jrnl today, and I was wondering, what happens when I export to YAML?
```
Uncovered while reviewing the tests in PR #824.

### Background/discussion
The YAML export's history is rooted in Day One compatibility (#159, #773, #349). That's fine, and it looks like we may need to preserve this behavior as a Day One export, but calling it a YAML export means that anybody expecting to extract/transform/load their jrnl anywhere else via YAML is going to have a bad time.

Maybe we rename the export type to something more appropriate to its intended function. Or, if the original target of this export can take proper YAML, then maybe we rework it. @MinchinWeb, any thoughts?

**Comment by wren** [comment]
Yup! I'm in favor of renaming it to clarify that it's DayOne-related. 

@MinchinWeb Your opinion?

**Comment by MinchinWeb** [comment]
The YAML export is designed to support DayOne (although I hope all exporters will do so), but it's actual roots are a modification of the Markdown exporter to include the entry metadata in a format useful as an input format for [Pelican](https://github.com/getpelican/pelican), a static site generator. This is why the files export with a `.md` file extension.

I'm fine with renaming it, but a more appropriate name would be along the lines of "Markdown with YAML front matter" or "Pelican Markdown"; i.e. nothing to do with DayOne.

---

"stared" appears to be an spelling mistake from when the exporter was [first added](https://github.com/jrnl-org/jrnl/commit/f7cadfc9dd13c5809f7049ea6e3176a3f08e67c6); no reason to keep the error.

**Comment by MinchinWeb** [comment]
The (original) discussion about why the exporter is named the why it is is found in this pull request --> https://github.com/jrnl-org/jrnl/pull/258

**Comment by micahellison** [comment]
This was fixed by #907
